spec/forms/creator_settings_form_spec.rb

```diff
@@ -0,0 +1,53 @@
+require "rails_helper"
+
+RSpec.describe CreatorSettingsForm, type: :model do
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:community_name) }
+    it { is_expected.to validate_presence_of(:primary_brand_color_hex) }
+  end
+
+  describe "attributes" do
+    it "has the correct attribute names" do
+      expect(described_class.attribute_names).to eq(%w[checked_code_of_conduct checked_terms_and_conditions
+                                                       community_name invite_only_mode logo
+                                                       primary_brand_color_hex public])
+    end
+
+    it "sets default values", :aggregate_failures do
+      attributes = described_class.new.attributes
+      expect(attributes["checked_code_of_conduct"]).to be(false)
+      expect(attributes["checked_terms_and_conditions"]).to be(false)
+      expect(attributes["invite_only_mode"]).to be(true)
+      expect(attributes["primary_brand_color_hex"]).to eq("#3b49df")
+      expect(attributes["public"]).to be(true)
+    end
+
+    it "updates default values", :aggregate_failures do
+      attributes = described_class.new(
+        primary_brand_color_hex: "#0a0a0a",
+        invite_only_mode: false,
+      ).attributes
+      expect(attributes["primary_brand_color_hex"]).to eq("#0a0a0a")
+      expect(attributes["invite_only_mode"]).to eq(false)
+    end
+  end
+
+  describe "#save" do
+    it "saves the updated attributes to the correct Settings values", :aggregate_failures do
+      creator_settings_form = described_class.new(
+        checked_code_of_conduct: true,
+        checked_terms_and_conditions: true,
+        community_name: "Climbing Life",
+        invite_only_mode: false,
+        public: false,
+        primary_brand_color_hex: "#a81adb",
+      )
+
+      expect(creator_settings_form.valid?).to be(true)
+      expect(Settings::Community.community_name).to eq("Climbing Life")
```

Comment:
@citizen428 I was wondering if you had some insights on this test that I'm struggling with please. This test block (with the Settings updates) are failing.

The code in the console is working as expected i.e. When we have a valid form object and we call `.save` on it, it updates all the Settings values. I see the expected results in my rails console if I test as follows:

```
c = CreatorSettingsForm.new(
        checked_code_of_conduct: true,
        checked_terms_and_conditions: true,
        community_name: "Climbing Life",
        invite_only_mode: false,
        public: false,
        primary_brand_color_hex: "#a81adb",
      )

c.valid?
# returns true

c.save
```

Now if I check the Settings values:
```
Settings::UserExperience.primary_brand_color_hex
# returns "a81adb"

Settings::UserExperience.public
# returns true
```

However, when I run this spec, the Settings values do not update. I know that on a model we'd usually do a `reload`, do you know how we'd check for the updated values in this case?

Reply:
@Ridhwana Reading this it seems like you're not actually saving the form in the spec. So I added an explicit call to `creator_settings_form.save` after the `expect(creator_settings_form.valid?).to be(true)` assertion and that did fix all but one spec. Not sure why `public` is not updating yet.

Reply:
BTW, regarding the problem with `true`, that also shows in your console example above:

```ruby
...
public: false,

Settings::UserExperience.public
# returns true
```

Reply:
Found the culprit! In `./spec/rails_helper.rb` we have the following:

```ruby
    allow(Settings::UserExperience).to receive(:public).and_return(true)
```

This is a leftover from the profile migration because otherwise we'd have needed to change a lot of specs. 

I took the liberty of pushing the fix directly to this branch, hope that's ok.